src/sidebar/test/service-url-test.js

```diff
@@ -1,20 +1,166 @@
 'use strict';
 
-var serviceUrl = require('../service-url');
+var proxyquire = require('proxyquire');
 
-describe('serviceUrl', function () {
-  var service;
+/** Return a fake annotationUI object. */
+function fakeAnnotationUI() {
+  var links = null;
+  return {
+    updateLinks: function(newLinks) {
+      links = newLinks;
+    },
+    getState: function() {
+      return {links: links};
+    },
+  };
+}
 
-  beforeEach(function () {
-    service = serviceUrl({serviceUrl: 'https://test.hypothes.is/'});
+function createServiceUrl(linksPromise) {
+  var replaceURLParams = sinon.stub().returns(
+    {url: 'EXPANDED_URL', params: {}}
+  );
```

Comment:
`replaceURLParams` could be simpler from the point of view of `serviceUrl` if it just returned the expanded URL (only) and raised an error if there were unused params. That way `serviceUrl` wouldn't have to check for unused params and format and throw the error itself. (And it wouldn't have to catch the error either, since it _wants_ to throw it.) But also this mock would just be `var replaceURLParams = sinon.stub().returns('EXPANDED_URL')` or, when testing what happens when an error is thrown, `var replaceURLParams = sinon.stub().throws(new Error())`. I haven't looked into how `replaceURLParams` is used elsewhere, though.